### DISCUSSION TITLE
Any solution for changing inputType fields name?

### DISCUSSION THREAD

**Discussion by arashi-dev** [MichalLytek/type-graphql#1153] [Q&A] [answered]
Hello

I am trying to change all the fields with name `_id` to `id` as I am using mongodb but I don't want my users to know that for security issues.
It is easy to change the field names in `ObjectTypes` but looks like we cannot change names in inputs.

Is there any solution for that? like a middleware? 
because half of my project is already done and it will takes a lot of time to change the names manually in the logics I wrote.

**Comment by arashi-dev** [comment]
I'm not sure if it is a good solution or not, but here is what I did

```ts
const Ref = (field: string) => {
    return (target: any, propertyKey: string) => {
        Object.defineProperty(target, propertyKey, {
            set: function (x) { this[field] = x },
            get: function () { return this[field] },
        });
        return target
    }
}

@InputType()
export class UserTestInput {
    @Ref("_id") <============== Ref decoration
    @Field(() => ObjectId, { nullable: true })
    id?: ObjectId

    _id: ObjectId
}
```

This decoration will replace the field by a getter and a setter with its own name (`id` in this case) and gets a name as an argument (`_id` in this example) and change the value of `_id` whenever we use the `id` setter.

this might have bugs, I appreciate if you check it and comment what you think about it

**Comment by MichalLytek** [comment, marked as answer]
You should be able to use getters and setters for aliasing properties, so that it's `id` outside and `_id` inside.
```ts
@InputType()
export class UserCreateInput {
  balance!: number;

  @Field(_type => Float, {
    nullable: false
  })
  get accountBalance() {
    return this.balance;
  }

  set accountBalance(balance: number) {
    this.balance = balance;
  }
}
```

**Reply by arashi-dev** [reply]
Thanks, so I will use the decoration I wrote as it is using this getter/setter approach
